.pre-commit-config.yaml

```diff
@@ -1,46 +1,18 @@
 repos:
- - repo: https://github.com/Quantco/pre-commit-mirrors-black
-   rev: 23.9.1
-   hooks:
-     - id: black-conda
-       additional_dependencies: [flake8-docstrings, flake8-rst-docstrings]
-       args:
-         - --safe
-         - --target-version=py39
- - repo: https://github.com/Quantco/pre-commit-mirrors-flake8
-   rev: 6.1.0
-   hooks:
-    - id: flake8-conda
-      additional_dependencies: [
-          -c,
-          conda-forge,
-          flake8-bugbear=21.4.3,
-          flake8-builtins=1.5.3,
-          flake8-comprehensions=3.5.0,
-          flake8-docstrings=1.6.0,
-          flake8-print=4.0.0,
-          pep8-naming=0.11.1,
-          python<3.12,
-        ]
- - repo: https://github.com/Quantco/pre-commit-mirrors-isort
-   rev: 5.12.0
-   hooks:
-    - id: isort-conda
-      additional_dependencies: [toml]
- - repo: https://github.com/Quantco/pre-commit-mirrors-mypy
-   rev: "1.5.1"
-   hooks:
-    - id: mypy-conda
-      additional_dependencies:
-       - python=3.9
- - repo: https://github.com/Quantco/pre-commit-mirrors-pyupgrade
-   rev: 3.15.0
-   hooks:
-    - id: pyupgrade-conda
-      args: [--py39-plus]
- - repo: https://github.com/Quantco/pre-commit-mirrors-cython-lint
-   rev: 0.15.0
-   hooks:
-    - id: cython-lint-conda
-      args: [--no-pycodestyle]
-    - id: double-quote-cython-strings-conda
+  - repo: https://github.com/Quantco/pre-commit-mirrors-ruff
+    rev: 0.1.3
+    hooks:
+      - id: ruff-conda
+      - id: ruff-format-conda
+  - repo: https://github.com/Quantco/pre-commit-mirrors-mypy
+    rev: "1.5.1"
+    hooks:
+      - id: mypy-conda
+        additional_dependencies:
+          - python=3.9
```

Comment:
```suggestion
```
Let's use this opportunity to get rid of this here and instead specify
```
[tool.mypy]
python_version = '3.9'
```
in `pyproject.toml`. The Python version that's used to run `mypy` doesn't have to be the same as the Python version that we're targeting.